### PR TITLE
Add edge unit tests for modules

### PR DESCRIPTION
**Pull request by saimonation**
**Add unit tests for edge modules:**

1. Asynchronous io (AIO)
2. FTP
3. SMB
4. NFS
5. AFP
6. NTP
7. Licenses
8. Directory Services
9. Users
10. Volumes
11. Config (hostname, location)
12. Rsync
13. Caching Gateway

### CODE REVIEW

File: tests/ut/test_edge_aio.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    def _get_cifs_object(self, enable_aio):
+        robust_mutexes, aio_read_threshold, aio_write_threshold = self._enable_aio if enable_aio else self._disable_aio
+        cifs_param = Object()
+        cifs_param.robustMutexes = robust_mutexes
+        cifs_param.aioReadThreshold = aio_read_threshold
+        cifs_param.aioWriteThreshold = aio_write_threshold
+        return cifs_param
+
+    def _assert_equal_objects(self, expected_param, actual_param):
```
Comment (ygalblum) [inline tests/ut/test_edge_aio.py]: You should move this to the SDK's base unittest class (since you implement it again in test_edge_directory_service

File: tests/ut/test_edge_directory_service.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        return cifs_param
+
+    def _get_workgroup_param(self):
+        cifs_param = Object()
+        cifs_param.type = "workgroup"
+        cifs_param.workgroup = self._workgroup
+        cifs_param.domain = None
+        return cifs_param
+
+    def _assert_equal_objects(self, expected_param, actual_param):
```
Comment (ygalblum) [inline tests/ut/test_edge_directory_service.py]: Should be moved to the base class

Comment (saimonation) [reply tests/ut/test_edge_directory_service.py]: Thanks @ygalblum . Moved **_assert_equal_objects** to **base.py**
